sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/io/CountingSourceTest.java

```diff
@@ -150,6 +153,40 @@ public void testUnboundedSourceTimestamps() {
   }
 
   @Test
+  public void testUnboundedSourceWithPeriod() {
+    Pipeline p = TestPipeline.create();
+
+    Duration period = Duration.millis(2);
```

Comment:
It almost certainly returns false - it should not take 2 seconds on modern hardware to generate 1000 longs without an explicit wait. On my machine, the testUnboundedSource test runs in about half a second, and this takes >2